### ISSUE TITLE
ngMask doesn't work properly when model is dynamic

### ISSUE THREAD

**Issue by cjfringe** [candreoliveira/ngMask#81]
We are trying to build a form dynamically, using ng-repeat. To do this, the model is set to a value tied to the id of the element in the loop. NgMask seems to be processing before the model is rendered, and is interpreting all fields as having the same model name. This is causing all fields to have the same mask applied, even though different masks are set for each field.

    <fieldset class="field" data-ng-repeat="field in vm.fieldsets.fields">
        <input
            id="{{field.id}}"
            type="{{field.type}}"
            data-ng-required="field.required"
            data-ng-model="vm.form[field.id]"
            placeholder="{{field.placeholder}}"
            name="{{field.name}}"
            data-mask="{{field.mask}}"
        >
    </fieldset>

In other words, it looks like ngMask is thinking the model for all fields is literally "vm.form[field.id]", not "vm.form['field1']", "vm.form['field2']", etc. Angular is interpreting the models correctly.

Any help resolving this issue would be appreciated!

**Comment by aymone** [comment]
I have the same problem here, any update for this issue?

**Comment by fabiowitt** [comment]
Nothing here?

**Comment by aaronroberson** [comment]
@cjfringe The fix for this has landed in v3.1.1.

**Comment by JMurilloM** [comment]
i have the same problem and i have a v3.1.1. any update?

**Comment by xxxalz** [comment]
I have the same problem

**Comment by lucis** [comment]
This issue should be reopened.

**Comment by erdahuja** [comment]
Same issue. Please help. browser crashes

**Comment by cait08** [comment]
I am having the same issue.
